cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/CreateCommand.java

```diff
@@ -0,0 +1,340 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.packerina.cmd;
+
+
+import org.ballerinalang.launcher.BLauncherCmd;
+import org.wso2.ballerinalang.util.RepoUtils;
+import picocli.CommandLine;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.AccessDeniedException;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.ballerinalang.packerina.cmd.Constants.CREATE_COMMAND;
+
+/**
+ * New command for creating a ballerina project.
+ */
+@CommandLine.Command(name = CREATE_COMMAND, description = "Create a new Ballerina project")
+public class CreateCommand implements BLauncherCmd {
+
+    private Path userDir;
+    private PrintStream errStream;
+    private static FileSystem jarFs;
+    private static Map<String, String> env;
+
+    @CommandLine.Parameters
+    private List<String> argList;
+
+    @CommandLine.Option(names = {"--help", "-h"}, hidden = true)
+    private boolean helpFlag;
+
+    @CommandLine.Option(names = {"--template", "-t"})
+    private String template = "main";
+
+    @CommandLine.Option(names = {"--list", "-l"})
+    private boolean list = false;
+
+    public CreateCommand() {
+        userDir = Paths.get(System.getProperty("user.dir"));
+        errStream = System.err;
+        initJarFs();
+    }
+
+    public CreateCommand(Path userDir, PrintStream errStream) {
+        this.userDir = userDir;
+        this.errStream = errStream;
+        initJarFs();
+    }
+
+    private void initJarFs() {
+        URI uri = null;
+        try {
+            uri = CreateCommand.class.getClassLoader().getResource("create_cmd_templates").toURI();
+            if (uri.toString().contains("!")) {
+                final String[] array = uri.toString().split("!");
+                if (null == jarFs) {
+                    env = new HashMap<>();
+                    jarFs = FileSystems.newFileSystem(URI.create(array[0]), env);
+                }
+            }
+        } catch (URISyntaxException e) {
+            throw new AssertionError();
+        } catch (IOException e) {
+            throw new AssertionError();
+        }
```

Comment:
Since the body of the two catch clauses are same we can change this to `URISyntaxException | IOException`